pykafka/producer.py

```diff
@@ -231,14 +231,38 @@ def _setup_owned_brokers(self):
                     self, partition.leader)
         return queued_messages
 
-    def stop(self):
-        """Mark the producer as stopped"""
-        self._running = False
+    def _stop_owned_brokers(self):
         self._wait_all()
         if self._owned_brokers is not None:
             for owned_broker in self._owned_brokers.values():
                 owned_broker.stop()
 
+    def get_queue_readers(self):
```

Comment:
I'm actually in favor of keeping this function inside `stop()` for now since it's only used there.